Main Cleanup of Framework release notes

sf1919 opened the pull request:
**Description of work.**

The main refactoring the the Framework release notes ahead of release of 6.2. Includes:-

- Adding/fixing links
- Updating titles and headings (including removing obsolete ones)
- Re-ordering information
- Spell checking
- Adding a screenshot

**To test:**

- Check that the links work
- Are the images visible and clearly illustrate the new feature/improvement?
- Grammar/Spelling check

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Refactoring of Framework release notes

Updated links, removed obsolete headings, added screenshots, spell check

sf1919 committed:
Further tidying of release notes

Reordered SliceViewer bugfixes list, revised note for IPeakFunction.intensityError, grammar check

DanielMurphy22 commented on docs/source/release/v6.2.0/framework.rst:
```diff
@@ -86,15 +81,13 @@ Bugfixes
-- Fixed bug in :ref:`algm-ConvertToMDMinMaxLocal` where wrong min max calculated if the workspace includes monitor spectra or spectra without any detectors
+- Fixed cursor tracking from getting stuck and displaying incorrect signals when viewing MDHistogram workspaces in :ref:`sliceviewer`.
 - Fix bug in :ref:`CalculateMultipleScattering <algm-CalculateMultipleScattering>` where detector position was incorrectly determined on a workspace where the workspace index didn't match the detector
-  index eg if the workspace was loaded with SpectrumMin specified to exclude some monitors
+  index e.g. if the workspace was loaded with ``SpectrumMin`` specified to exclude some monitors.
+- Fixed bug in :ref:`algm-ConvertToMDMinMaxLocal` where wrong min max calculated if the workspace includes monitor spectra or spectra without any detectors.
+- Added parser for input Names to :ref:`algm-CreateMDHistoWorkspace` to allow inputs such as `Names='[H,0,0],[0,K,0],[0,0,L]'`.
```
Can I ask for these 3 bullet points on CalculateMultipleScattering, ConvertToMDMinMaxLocal, CreateMDHistoWorkspace to be moved up to the Algorithms/Bugfixes section? One of them is mine, but they aren't really sliceviewer related.

DanielMurphy22 reviewed (changes requested):
Links, language and formatting are all great! Just a few small points:
- The image I meant for the Framework was this one (Microphone not tube), sorry for not being clear!
![MeshPlotHelp-2](https://user-images.githubusercontent.com/55980573/131828812-c08a036f-2710-4736-8633-b42627e53cc3.png)

- In [new line 79](https://github.com/mantidproject/mantid/pull/32340/files#diff-c6f0a92d777542d361e5213967a96e5ecb492802d83fb8391ac0ede58c529019R79) can we change it to something like
```
- :ref:`LoadNexusLogs <algm-LoadNexusLogs>` now creates a warning message for logs that are poorly formed and the other logs are loaded. Previously it stopped loading logs at that point.
```

sf1919 committed:
Revisions after review

Changed screenshot, reworded an entry, amalgamated Algorithms into single section and moved some SliceViewer bugfixes to Algorithms

DanielMurphy22 reviewed (dismissed):
Looks great! Well done on spotting the 2 Algorithm sections!

gvardany commented on docs/source/release/v6.2.0/framework.rst:
```diff
@@ -5,96 +5,81 @@ Framework Changes
+New Features
+############
+
+- :ref:`SaveAscii <algm-SaveAscii>` will no longer throw an error if `WriteXErrors` is requested, but there are no `Dx` data present in the workspace.
```
Could you move this item down to the improvements?

sf1919 committed:
Moved note on SaveAscii to Improvements
